packages/dev/core/src/AudioV2/abstractAudio/audioEngineV2.ts

```diff
@@ -0,0 +1,171 @@
+
+    private _defaultMainBus: Nullable<MainAudioBus> = null;
+
+    protected constructor() {
+        Instances.push(this);
+    }
+
+    /**
+     * The elapsed time since the audio engine was started, in seconds.
+     */
+    public abstract get currentTime(): number;
+
+    /**
+     * The default main bus that will be used for audio buses and sounds if their `outBus` option is not set.
+     * @see {@link IAudioBusOptions.outBus}
+     * @see {@link IAbstractSoundOptions.outBus}
+     */
+    public get defaultMainBus(): Nullable<MainAudioBus> {
+        if (this._mainBuses.size === 0) {
+            return null;
+        }
+
+        if (!this._defaultMainBus) {
+            this._defaultMainBus = Array.from(this._mainBuses)[0];
+        }
+
+        return this._defaultMainBus;
+    }
+
+    /**
+     * `true` if the engine is a WebAudio engine; otherwise `false`.
+     */
+    public abstract get isWebAudio(): boolean;
+
+    /**
+     * The spatial audio listener properties for the audio engine.
+     * - Each audio engine has exactly one listener.
+     */
+    public abstract get listener(): AbstractSpatialAudioListener;
+
+    /**
+     * The main output node.
+     * - This is the last node in the audio graph before the audio is sent to the speakers.
+     */
+    public abstract get mainOut(): AbstractAudioNode;
+
+    /**
+     * The current state of the audio engine.
+     */
+    public abstract get state(): AudioEngineV2State;
+
+    /**
+     * The output volume of the audio engine.
+     */
+    public abstract get volume(): number;
+    public abstract set volume(value: number);
+
+    /**
+     * Releases associated resources.
+     */
+    public dispose(): void {
+        if (Instances.includes(this)) {
+            Instances.splice(Instances.indexOf(this), 1);
+        }
+
+        const nodeIt = this._nodes.values();
+        for (let next = nodeIt.next(); !next.done; next = nodeIt.next()) {
+            next.value.dispose();
+        }
+
+        this._mainBuses.clear();
+        this._nodes.clear();
+
+        this._defaultMainBus = null;
+    }
+
+    /**
+     * Checks if the specified format is valid.
+     * @param format The format to check as an audio file extension like "mp3" or "wav".
+     * @returns `true` if the format is valid; otherwise `false`.
+     */
+    public abstract formatIsValid(format: string): boolean;
+
+    /**
+     * Pauses the audio engine if it is running.
+     * @returns A promise that resolves when the audio engine is paused.
+     */
+    public abstract pause(): Promise<void>;
+
+    /**
+     * Resumes the audio engine if it is not running.
+     * @returns A promise that resolves when the audio engine is running.
+     */
+    public abstract resume(): Promise<void>;
+
+    /**
+     * Unlocks the audio engine if it is locked.
+     * - Note that the returned promise resolves immediately if the audio engine is already unlocked.
```

Comment:
But will still be a microtask, so it will still be resolved after other code has already ran.
I guess resolves immediately is acceptable in javascript lingo though

Reply:
Maybe return `void | Promise<void>` instead? ...or is this only a docs issue? How would you put it to be more clear?